tests/test_objective_funs.py

```diff
@@ -655,6 +660,180 @@ def test(coil, grid=None):
         test(nested_coils, grid=nested_grids)
 
 
+class TestQuadraticFlux:
+    """Tests quadratic flux."""
+
+    def get_grids(self, eq, field, N_grid=15, M_grid=10):
+        """Compute grids used for all of the tests."""
+        field_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=M_grid,
+            N=N_grid,
+            NFP=field.NFP if not isinstance(field, ToroidalMagneticField) else 1,
+            sym=False,
+        )
+        eval_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=eq.M_grid,
+            N=N_grid,
+            NFP=eq.NFP,
+            sym=False,
+        )
+
+        source_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=M_grid,
+            N=N_grid,
+            NFP=int(eq.NFP),
+            sym=False,
+        )
+
+        return field_grid, eval_grid, source_grid
+
+    def make_nonaxisym_field(self, eq):
+        """Makes a nonaxisymmetric field depending on the given eq."""
+        surf = eq.surface.copy()
+        surf.R_lmn = put(surf.R_lmn, surf.R_basis.get_idx(M=1, N=0), 2)
+        surf.Z_lmn = put(surf.Z_lmn, surf.Z_basis.get_idx(M=-1, N=0), -2)
+        nonaxisym_field = FourierCurrentPotentialField.from_surface(
+            surface=surf,
+            Phi_mn=np.array([1e4]),
+            modes_Phi=np.array([[1, 0]]),
+            G=1e4,
+            I=0,
+            sym_Phi="sin",
+        )
+
+        return nonaxisym_field
+
+    @pytest.mark.unit
+    def test_quadratic_flux(self):
+        """Test calculation of quadratic flux on the boundary."""
+        t_field = ToroidalMagneticField(1, 1)
+
+        # test that torus (axisymmetric) Bnorm is exactly 0
+        eq = load("./tests/inputs/vacuum_circular_tokamak.h5")
+        obj = QuadraticFlux(eq, t_field)
+        obj.build(eq, verbose=2)
+        f = obj.compute(field_params=t_field.params_dict)
+        np.testing.assert_allclose(f, 0, rtol=1e-14, atol=1e-14)
+
+        # test nonaxisymmetric surface
+        eq = desc.examples.get("precise_QA", "all")[0]
+        eq.change_resolution(4, 4, 4, 8, 8, 8)
+
+        __, eval_grid, source_grid = self.get_grids(eq, t_field)
+
+        obj = QuadraticFlux(
+            eq,
+            t_field,
+            eval_grid=eval_grid,
+            source_grid=source_grid,
+        )
+        Bnorm = t_field.compute_Bnormal(
+            eq.surface, eval_grid=eval_grid, source_grid=source_grid
+        )[0]
+        obj.build(eq)
+        dA = eq.compute("|e_theta x e_zeta|", grid=eval_grid)["|e_theta x e_zeta|"]
+        f = obj.compute(field_params=t_field.params_dict)
+
+        np.testing.assert_allclose(f, Bnorm * dA, atol=1e-3)
+
+    @pytest.mark.unit
+    def test_quadratic_flux_with_eq_fixed(self):
```

Comment:
we can get rid of this test, as the eq fixed part is basically tested with the following test